app/(root)/posts/create-post/page.tsx

```diff
@@ -0,0 +1,104 @@
+"use client";
+
+import React, { useEffect } from "react";
+import { Loader2, X } from "lucide-react";
+
+import { zodResolver } from "@hookform/resolvers/zod";
+import {
+  SubmitHandler,
+  useForm,
+  useFieldArray,
+  Controller,
+} from "react-hook-form";
+import { useRouter } from "next/navigation";
+import toast from "react-hot-toast";
+
+import { CreateType, Resource, User } from "@prisma/client";
+import { IPostSchema, PostSchema } from "@/lib/validations/PostSchema";
+
+import { createPost } from "@/lib/actions/post.actions";
+import { Button } from "@/components/shared/ui";
+import BasicInformationPost from "@/components/post/BasicInformationPost";
+import Content from "@/components/post/Content";
+import Resources from "@/components/post/Resources";
+
+const CreatePost = ({ user }: { user: User & { resources?: Resource[] } }) => {
+  const router = useRouter();
+  const useFormHelpers = useForm<IPostSchema>({
+    defaultValues: {
+      title: "",
+      createType: CreateType.COMPONENT,
+      description: "",
+      codeEditor: "",
+      content: "",
+      steps: [],
+      learnings: [],
+      tags: [],
+      resources: [],
+    },
+    resolver: zodResolver(PostSchema),
+  });
+
+  const {
+    register,
+    watch,
+    handleSubmit,
+    formState: { isSubmitting, errors },
+  } = useFormHelpers;
+  const { control } = useFormHelpers;
+  const contentWatch = watch("content");
+
+  useEffect(() => {
+    console.log("contentWatch", contentWatch);
+  }, [contentWatch]);
+
+  const onSubmit: SubmitHandler<IPostSchema> = async (data) => {
+    try {
+      await createPost(data);
```

Comment:
This data is giving me a typescript error:

![Screenshot 2024-04-03 at 11 27 33 AM](https://github.com/DDVVPP/GitNote/assets/47732125/eb0dd1df-cd29-4b67-87e2-05e85be2d481)

I'm not sure why it thinks that codeEditor will be null, rather than undefined?